Microsoft.Quantum.Canon/Arrays/Arrays.qs

```diff
@@ -252,5 +253,52 @@ namespace Microsoft.Quantum.Canon {
 		return sliced;
 	}
 
+    /// # Summary
+    /// Returns an array padded at the tail with specified values up to a 
+    /// specified length.
+    ///
+    /// # Type Parameters
+    /// ## 'T
+    /// The type of the array elements.
+    ///
+    /// # Input
+    /// ## nElementsTotal
+    /// The length of the padded array. If this is positive, `inputArray`
+    /// is padded at the tail. If this is negative, `inputArray` is padded
+    /// at the head.
+    /// ## defaultElement
+    /// Default value to use for padding elements.
+    /// ## inputArray
+    /// Array whose values are at the head of the output array.
+    ///
+    /// # Output
+    /// An array `output` that is the `inputArray` padded at the tail
+    /// with `defaultElement`s until `output` has length `nElementsTotal`
+    ///
+    /// # Example
+    /// ```Q#
+    /// let array = [10; 11; 12];
+    /// // The following line returns [10; 12; 15; 2; 2; 2].
+    /// let output = PadTail(6, array, 2);
+    /// // The following line returns [2; 2; 2; 10; 12; 15].
+    /// let output = PadTail(-6, array, 2);
+    /// ```
+    function PadTail<'T>(nElementsTotal: Int, inputArray: 'T[], defaultElement: 'T) : 'T[] 
```

Comment:
It might make sense for `inputArray` to come last, since it will be the least likely element to be specified as a part of partial application. That way, the order of `PadTail`'s arguments follows roughly currying intuition.

Reply:
`inputArray' now comes last.